addons/stock/views/product_strategy_views.xml

```diff
@@ -15,24 +15,31 @@
                 <field name="product_id" string="Product"
                        options="{'no_create': True, 'no_open': True}"
                        readonly="context.get('single_product', False) or category_id"
-                       required="not category_id and not package_type_ids"
+                       placeholder="All Products"
                        force_save="1"/>
                 <field name="category_id" string="Product Category"
                        options="{'no_create': True, 'no_open': True}"
                        readonly="context.get('fixed_category', False) or product_id"
-                       required="not product_id and not package_type_ids"
+                       placeholder="All Categories"
                        force_save="1"/>
                 <field name="package_type_ids" string="Package type"
                        options="{'no_create': True, 'no_open': True}"
                        groups="stock.group_tracking_lot" widget="many2many_tags"
                        optional="show"/>
-                <field name="location_out_id"
+                <field name="location_out_id" string="Store to"
                        readonly="not location_in_id"
                        options="{'no_create': True}"
+                       invisible="sublocation == 'last_used'"
```

Comment:
The problem with this is that a rule can be applied to multiple products at the same tin time (if we use a product category or "All Products"). In that case, we can have multiple locations per rule.

Reply:
`location_out_id` by default will be equal to the `location_in_id` at the time of rule creation. Let's say we have a putaway rule for all products and all categories employing `last_used` strategy. Each product of those will have a last location depending on the last ingoing move with this product. So in this case, `location_out_id` doesn't hold any meaning in `last_used` strategy. Actually, we can even fallback to `location_in_id` if we couldn't find a previous move for a product (`location_in_id` and `location_out_id` will be equal anyway, but just for clarity).

Reply:
@ABE-Mark45 You don't get it, you missed the "*the field is invisible but not empty if previously set*" of my first message...
Try that:
- Create a new product;
- Create a putaway rule, for the created product or its category (like you prefer), define a "Store to" location different than its `location_in_id`, "Sublocation" set to "No" and save it (the putaway record must be created);
- Edit your putaway rule and only edit the "Sublocation" from "No" to "Last Used" -> The "Store to" field is now invisible.
- Create a receipt for this product, location dest = putaway's `location_in_id` and confirm it -> The move line destination is the putaway "Store to" location.

Making the field invisible will not magically empty the field or reset it to its default value. Since this field is still used as the fallback when no previous location exists for the product, I think it's a bad idea to make it invisible

Reply:
@svs-odoo I agree and this is why I was saying that we don't actually need to use `location_out_id` in the `last_used` strategy. We can use here the `location_in_id` as a fallback, then we can make `Store to` invisible in this strategy to avoid confusing the user with redundant information. We can replace [this line](https://github.com/odoo/odoo/blob/f19843a3db39cc16497832379a059713845ddee8/addons/stock/models/product_strategy.py#L153) with
```python
location_out = putaway_rule.location_in_id
```

Reply:
@ABE-Mark45 I think not using `location_out_id` at all in case we're using `last_used` as sublocation makes sense too (that said, should not be in this line but after the condition checking if `last_used` is selected since we still want to use `location_out_id` in other case).
In fact, there is another solution who could make sense: if we're using `last_used` as sublocation, the last known location is searched only in `location_out_id` and its sublocations, but I think it requires more development so maybe for another task but not this one.

For now, I'm still thinking it's wrong to hide the field while using it, so it could be useful to ask @thomasdeleval what solution he prefers since it's more a functional preference